src/audio/module_adapter/module_adapter.c

```diff
@@ -548,6 +548,10 @@ int module_adapter_params(struct comp_dev *dev, struct sof_ipc_stream_params *pa
 			return ret;
 	}
 
+#if CONFIG_IPC_MAJOR_4
+	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, params);
+	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, mod->stream_params);
```

Comment:
for my understanding: why is this needed? Is this fixing a bug?

Reply:
Yes, now that the copier is not using the comp_driver, its params wont get called until prepare. So the params if we wont do this and components that do not use the module adapter yet will fail

Reply:
does that mean that that before this commit copier.c is broken, which breaks bisection? Should this commit be merged with https://github.com/thesofproject/sof/pull/7890/commits/6bcaf899cde497db312603408877401d24f35768 which comes before it?